### DISCUSSION TITLE
Question: mandatory AGTextureUnit index

### DISCUSSION THREAD

**Discussion by BioRyajenka** [korlibs/korge#1171] [Q&A] [answered]
Hi, 
With this new requirement, what kind of indices should I use when creating AGTextureUnit?

Should it always be < BB_MAX_TEXTURES? 
Should always start with 6? (saw in a couple of classes in korge)
Can it be made optional somehow, from within the engine?

My case:
I am generating tilemaps. 1 tilemap = 1 chunk, so I have to create many of them, and each tex should have unique index, as I understand it. If I have 1000 tilemaps, I will need at least 1000 different texture numbers. Is it ok?
Obviously I don't draw 1000 tilemaps at the same time, only 4 of them.

**Comment by soywiz** [comment, marked as answer]
That index will be probably removed in next alpha and will be automatically handled. AGTextureUnit itself will be probably removed too.

**Reply by soywiz** [reply]
Not required anymore after this commit: https://github.com/korlibs/korge/commit/dd08592e481ad19c7ac7b6bf69c99b7e6269cdd1

**Reply by BioRyajenka** [reply]
Thanks!
